core/src/main/java/org/bitcoinj/core/PeerGroup.java

```diff
@@ -243,6 +243,8 @@ public void onPeerDisconnected(Peer peer, int peerCount) {
     /** The default timeout between when a connection attempt begins and version message exchange completes */
     public static final int DEFAULT_CONNECT_TIMEOUT_MILLIS = 5000;
     private volatile int vConnectTimeoutMillis = DEFAULT_CONNECT_TIMEOUT_MILLIS;
+    
+    private boolean bloomFilteringEnabled = true;
```

Comment:
Make this volatile please, or set it under the peer group lock. Otherwise we're violating the JMM. 

Also you may as well check it directly where the decisions are made and avoid the method call (it's not that big a deal on any reasonable JVM but it can't hurt to access directly).